File: blind75/72_meeting_rooms/meeting_rooms.go
package meeting_rooms

import "sort"

// CanAttendMeetings determines if a person can attend all meetings.
// Time: O(n log n), Space: O(1)
func CanAttendMeetings(intervals [][]int) bool {
	if len(intervals) <= 1 {
		return true
	}

	// Sort intervals by start time
	sort.Slice(intervals, func(i, j int) bool {
		return intervals[i][0] < intervals[j][0]
	})

	// Check for overlaps in consecutive intervals
	for i := 1; i < len(intervals); i++ {
		if intervals[i][0] < intervals[i-1][1] {
			return false // Overlap found
		}
	}

	return true
}

File: blind75/72_meeting_rooms/meeting_rooms.py
from typing import List


def can_attend_meetings(intervals: List[List[int]]) -> bool:
    """
    Determines if a person can attend all meetings.
    Time: O(n log n), Space: O(1)

    Args:
        intervals: List of meeting time intervals [start, end]

    Returns:
        True if person can attend all meetings, False otherwise
    """
    if len(intervals) <= 1:
        return True

    # Sort intervals by start time
    intervals.sort(key=lambda x: x[0])

    # Check for overlaps in consecutive intervals
    for i in range(1, len(intervals)):
        if intervals[i][0] < intervals[i - 1][1]:
            return False  # Overlap found

    return True
